index.php

```diff
@@ -4,63 +4,107 @@
 
 function prepDirectoryPath($file) {
 
-    $path = explode('/', $file);
+    //ensure that the file is not empty
+    if($file!=='' &&!empty($file) && isset($empty))
+    {
+        $path = explode('/', $file);
 
-    // Pop off the file name and rebuild the full path
-    array_pop($path);
-    $fullPath = '';
+        // Pop off the file name and rebuild the full path
+        array_pop($path);
+        $fullPath = '';
 
-    // If the directory already exists, bail now
-    if (!is_dir(LOCAL_PATH . '/' . implode('/', $path))) {
+        // If the directory already exists, bail now
+        if (!is_dir(LOCAL_PATH . '/' . implode('/', $path))) {
 
-        // Verify and create each section as necessary
-        foreach ($path as $directory) {
-            if (strlen($directory) > 0) {
-                $fullPath .= '/' . $directory;
-                if (!is_dir(LOCAL_PATH . '/' . $fullPath)) {
-                    mkdir(LOCAL_PATH . '/' . $fullPath);
+            // Verify and create each section as necessary
+            foreach ($path as $directory) {
+                if (strlen($directory) > 0) {
+                    $fullPath .= '/' . $directory;
+                    if (!is_dir(LOCAL_PATH . '/' . $fullPath))
+                    {
+                        mkdir(LOCAL_PATH . '/' . $fullPath);
+                    }
                 }
             }
+
         }
+    }
 
+    else
+    {
+        return false;
     }
 
+
 }
 
 function getImage($fileName, $source) {
     $retVal = false;
 
-    // Verify the image is okay first
-    $url = $source . $fileName;
-    $headers = @get_headers($url, 1);
+    //verify that the fields are not empty
+    if($fileName!=='' && !empty($fileName) && isset($fileName) && $source!=='' && !empty($source) && isset($source))
+    {
+        try
```

Comment:
`file_get_contents` will throw a warning which cannot be caught and `get_headers` will never throw an exception. It's expected that `file_get_contents` may fail due to service outages or whatever and the suppression was added to keep from spamming logs. Revert to old code.